spam/03_spam_data_slicing_tutorial.py

```diff
@@ -352,93 +239,166 @@ def textblob_polarity(x):
 sfs = [short_link] + extra_sfs
 slice_names = [sf.name for sf in sfs]
 
+# %% [markdown]
+# Let's see how the `sklearn` model we learned before performs on these new slices!
+
 # %%
 applier = PandasSFApplier(sfs)
 S_test = applier.apply(df_test)
 
-slice_scorer = SliceScorer(scorer, slice_names)
-slice_scorer.score(
+slice_scorer = SliceScorer(slice_names, metrics=["accuracy", "f1"])
+slice_scorer.score_slices(
     S=S_test, golds=Y_test, preds=preds_test, probs=probs_test, as_dataframe=True
 )
 
 # %% [markdown]
-# ## 5. Improve slice performance
+# Looks like they do well — we'll want to monitor these to make sure performance changes on one don't hurt another.
+
+# %% [markdown]
+# ## 3. Improve slice performance
+#
+# In the following section, we demonstrate a modeling approach that we call _Slice-based Learning,_ which improves performance with slice-specific representation learning.
+# Intuitively, we'd like to model to learn *representations that are better suited to handle examples in this slice*.
+# In our approach, we model each slice as a separate "expert task" in the style of [multi-task learning](https://github.com/snorkel-team/snorkel-tutorials/blob/master/multitask/multitask_tutorial.ipynb).
```

Comment:
Be clear what will be in this tutorial and what won't- so I would tack on to this sentence something after a semi-colon like: "for further details of how slice-based learning works under the hood, check out [the code?] or [arxiv link?]"